src/services/onchain-data.service.ts

```diff
@@ -249,6 +255,107 @@ export class OnchainDataService implements OnModuleInit
         return this.tokens;
     }
 
+    @Interval(60000)
+    public async fetchTradingCompetitionRanking(): Promise<void>
+    {
+        const swapEventABI: object = find(VexchangeV2PairABI, { name: "Swap" });
+        const pairContract: Connex.Thor.Account.Visitor = this.Connex.thor.account(this.rankingPairAddress);
+        const swapEvent: Connex.Thor.Account.Event =
+        pairContract.event(swapEventABI);
+
+        const swapFilter: Connex.Thor.Filter<"event", Connex.Thor.Account.WithDecoded> = swapEvent
+            .filter([])
+            .range({
+                unit: "block",
+                // Since every block is 10s, 8640 blocks will be 24h
+                from: this.Connex.thor.status.head.number - 8640 * 60,
+                // Current block number
+                to: this.Connex.thor.status.head.number,
+            });
+
+        let end: boolean = false;
+        let offset: number = 0;
+        const limit: number = 256;
+
+        // Need a while loop because we can only get
+        // up to 256 events each round using connex
+        while (!end)
+        {
+            const result: Connex.Thor.Filter.Row<
+                "event", Connex.Thor.Account.WithDecoded
+            >[] = await swapFilter.apply(offset, limit);
+
+            for (const transaction of result)
+            {
+                // eslint-disable-next-line
+                if (!this.pointsPerAddress[transaction.decoded.sender])
+                {
+                    this.pointsPerAddress[transaction.decoded.sender] = {
+                        points: ethers.constants.Zero,
+                    };
+                }
+
+                this.pointsPerAddress[transaction.decoded.sender].points = this.pointsPerAddress[
+                    transaction.decoded.sender
+                ].points
+                    .add(transaction.decoded.amount0In)
+                    .add(transaction.decoded.amount0Out);
+            }
+
+            if (result.length === limit) offset += limit;
+            else end = true;
```

Comment:
```suggestion
            if (result.length === limit) { offset += limit; }
            else { end = true; }
```

to be safer. We can change the other instance of this too in the `fetch` function